mcp85xx-p1020: add support for Aerohive HiveAP 330

herrbett opened the pull request:
- [x] must be flashable from vendor firmware
  - [ ] webinterface
  - [x] tftp
  - [x] other: serial - process is similar to Aerohive HiveAP 121
- [x] must support upgrade mechanism
  - [x] must have working sysupgrade
    - [x] must keep/forget configuration (if applicable)
      *think `sysupgrade [-n]` or `firstboot`*
  - [x] must have working autoupdate
    *usually means: gluon profile name must match image name*
    `aerohive-hiveap-330`
- [x] reset/wps button must return device into config mode
- [x] primary mac should match address on device label (or packaging) (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)
- wired network
  - [x] should support all network ports on the device
  - [x] must have correct port assignment (WAN/LAN)
- wifi (if applicable)
  - [x] association with AP must be possible on all radios
  - [x] association with 802.11s mesh must be working on all radios 
  - [x] ap/mesh mode must work in parallel on all radios
- led mapping
  - power/sys led (_critical, because led definitions are setup on firstboot only_)
    - [x] lit while the device is on
    - [x] should display config mode blink sequence 
(https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - radio leds - device does not have radio leds 
    - [ ] should map to their respective radio
    - [ ] should show activity
  - switchport leds - device does not have switchport leds
    - [ ] should map to their respective port (or switch, if only one led present) 
    - [ ] should show link state and activity
- outdoor devices only - no outdoor Device
  - [ ] added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`

mweinelt commented on targets/mpc85xx-p1020:
```diff
@@ -0,0 +1,3 @@
+device('aerohive-hiveap-330', 'hiveap-330', {
```
Add the vendor name as a comment, as seen in other target files.

mweinelt commented:
Add the subtarget name (p1020) to the commit message.

mweinelt commented:
Please add your Hive APs also to the docs https://github.com/freifunk-gluon/gluon/blob/master/docs/index.rst.

herrbett commented:
I added the comment with the vendor name in the taget file and also added the sub target name to to the commit message.

As the Aerohive AP-121 is also missing in the docs, I will open another PR and add both APs to the Docs

mweinelt commented:
As docs are a minor issue you can just add another commit to this PR.

herrbett commented:
Ok, I added the documentation commit to this PR.

mweinelt commented on docs/index.rst:
```diff
@@ -122,6 +122,11 @@ ar71xx-generic
+
+  - HiveAP 121
+  - HiveAP 330
+
```
They're not in ar71xx-generic, are they?

herrbett replied:
Sorry, fixed that. They are in their respective target now

mweinelt commented:
Thank you, merged.
